src/operator/single.ts

```diff
@@ -55,20 +52,25 @@ class SingleSubscriber<T> extends Subscriber<T> {
 
   protected _next(value: T): void {
     const predicate = this.predicate;
-    const currentIndex = this.index++;
-
+    this.index++;
     if (predicate) {
-      let result = tryCatch(predicate)(value, currentIndex, this.source);
-      if (result === errorObject) {
-        this.destination.error(errorObject.e);
-      } else if (result) {
-        this.applySingleValue(value);
-      }
+      this.__tryNext(value);
     } else {
       this.applySingleValue(value);
     }
   }
 
+  private __tryNext(value: T): void {
```

Comment:
No actually. I confess that I'm always a bit confused about the use of underbar across the codebase. Not sure why I used it here. I'll rename it to `tryNext()` and update this PR. ;)